### PR TITLE
Add product test launcher 'env describe' command

### PR DESCRIPTION
**Pull request by wendigo**
This PR adds `env inspect` command that works similar to `env up` but instead of creating and starting environment it just displays what environment consists of in following form:

```
./testing/trino-product-tests-launcher/bin/run-launcher env describe --environment singlenode --config config-hdp3 --
2021-05-28T15:57:20.032+0200    INFO    main    io.trino.tests.product.launcher.cli.EnvironmentDescribe Environment 'singlenode' containers:
+---------------+-----------------------------------------+------------------------------+-----+-------+----------------------------------------------------+
|     container |                                   image |                network alias | env | ports |                                        run command |
+---------------+-----------------------------------------+------------------------------+-----+-------+----------------------------------------------------+
|         tests |  ghcr.io/trinodb/testing/centos6-oj8:34 |                  tc-GTsMqLcW |     |       | bash -xeuc echo 'No command provided' >&2; exit 69 |
|               |                                         |                        tests |     |       |                                                    |
+---------------+-----------------------------------------+------------------------------+-----+-------+----------------------------------------------------+
| presto-master | ghcr.io/trinodb/testing/centos7-oj11:34 |                  tc-EhATM6Y5 |     |  5005 |         /docker/presto-product-tests/run-presto.sh |
|               |                                         | presto-master.docker.cluster |     |  8080 |                                                    |
|               |                                         |                presto-master |     |       |                                                    |
+---------------+-----------------------------------------+------------------------------+-----+-------+----------------------------------------------------+
| hadoop-master |  ghcr.io/trinodb/testing/hdp3.1-hive:34 |                  tc-1T1oA325 |     |  1180 |                           /usr/local/hadoop-run.sh |
|               |                                         |                hadoop-master |     |  8020 |                                                    |
|               |                                         |                              |     |  8042 |                                                    |
|               |                                         |                              |     |  8088 |                                                    |
|               |                                         |                              |     |  9000 |                                                    |
|               |                                         |                              |     |  9083 |                                                    |
|               |                                         |                              |     |  9864 |                                                    |
|               |                                         |                              |     |  9870 |                                                    |
|               |                                         |                              |     | 10000 |                                                    |
|               |                                         |                              |     | 19888 |                                                    |
|               |                                         |                              |     | 50070 |                                                    |
|               |                                         |                              |     | 50075 |                                                    |
+---------------+-----------------------------------------+------------------------------+-----+-------+----------------------------------------------------+
2021-05-28T15:57:20.087+0200    INFO    main    io.trino.tests.product.launcher.cli.EnvironmentDescribe Environment 'singlenode' file mounts:
+---------------+------+------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------+------+----------+
|     container | type |                                                                                                                   from |                                                                                        to | type |     size |
+---------------+------+------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------+------+----------+
|         tests | copy |                                                                                  classpath:docker/presto-product-tests |                                                              /docker/presto-product-tests |  dir | 188.78kB |
+---------------+------+------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------+------+----------+
| presto-master | copy |                                                                                  classpath:docker/presto-product-tests |                                                              /docker/presto-product-tests |  dir | 188.78kB |
| presto-master | copy |                                                       classpath:docker/presto-product-tests/conf/presto/etc/jvm.config |                                   /docker/presto-product-tests/conf/presto/etc/jvm.config | file |     656B |
| presto-master | copy |                                              classpath:docker/presto-product-tests/health-checks/trino-health-check.sh |                                                       /etc/health.d/trino-health-check.sh | file |     331B |
| presto-master | copy |                            /var/folders/kj/q89730fx15g7gs2lszthfp_40000gn/T/enable-java-debugger6646706103428298393.sh |                                             /docker/presto-init.d/enable-java-debugger.sh | file |     164B |
| presto-master | copy |                                        classpath:docker/presto-product-tests/common/standard/access-control.properties |                    /docker/presto-product-tests/conf/presto/etc/access-control.properties | file |      30B |
| presto-master | copy |                                                classpath:docker/presto-product-tests/common/standard/config.properties |                            /docker/presto-product-tests/conf/presto/etc/config.properties | file |     295B |
| presto-master | copy |                                                    classpath:docker/presto-product-tests/common/hadoop/hive.properties |                      /docker/presto-product-tests/conf/presto/etc/catalog/hive.properties | file |     532B |
| presto-master | copy |                               classpath:docker/presto-product-tests/common/hadoop/hive_with_external_writes.properties | /docker/presto-product-tests/conf/presto/etc/catalog/hive_with_external_writes.properties | file |     575B |
| presto-master | copy |                                    classpath:docker/presto-product-tests/common/hadoop/hive_timestamp_nanos.properties |      /docker/presto-product-tests/conf/presto/etc/catalog/hive_timestamp_nanos.properties | file |     290B |
| presto-master | copy |                                                 classpath:docker/presto-product-tests/common/hadoop/iceberg.properties |                   /docker/presto-product-tests/conf/presto/etc/catalog/iceberg.properties | file |     190B |
| presto-master | copy |                                              classpath:docker/presto-product-tests/common/standard/presto-init-hdp3.sh |                                                 /docker/presto-init.d/presto-init-hdp3.sh | file |     366B |
| presto-master | bind | /Users/mateuszgajewski/Projects/src/github.com/trinodb/trino/core/trino-server/target/trino-server-358-SNAPSHOT.tar.gz |                                                              /docker/presto-server.tar.gz | file | 582.36MB |
+---------------+------+------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------+------+----------+
| hadoop-master | copy |                                                                                  classpath:docker/presto-product-tests |                                                             /docker/presto-product-tests/ |  dir | 188.78kB |
| hadoop-master | copy |                                             classpath:docker/presto-product-tests/health-checks/hadoop-health-check.sh |                                                      /etc/health.d/hadoop-health-check.sh | file |     415B |
| hadoop-master | copy |                                                      classpath:docker/presto-product-tests/common/hadoop/hadoop-run.sh |                                                                  /usr/local/hadoop-run.sh | file |     458B |
| hadoop-master | copy |                                          classpath:docker/presto-product-tests/common/hadoop/apply-config-overrides.sh |                                           /etc/hadoop-init.d/00-apply-config-overrides.sh | file |     458B |
| hadoop-master | copy |                                                          classpath:docker/presto-product-tests/health-checks/health.sh |                                                                  /usr/local/bin/health.sh | file |     327B |
+---------------+------+------------------------------------------------------------------------------------------------------------------------+-------------------------------------------------------------------------------------------+------+----------+
```

### CODE REVIEW

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/EnvironmentInspect.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                    MountableFile mountableFile = file.getKey();
+                    Path mountedFilePath = Paths.get(mountableFile.getFilesystemPath());
+                    boolean isDirectory = Files.isDirectory(mountedFilePath);
+
+                    mountsTable.addRow(
+                            container.getLogicalName(),
+                            "copy",
+                            mountableFile.getDescription(),
+                            file.getValue(),
+                            isDirectory ? "dir" : "file",
```
Comment (losipiuk) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/EnvironmentInspect.java:167]: are file/dir only options? symlink?

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/EnvironmentInspect.java [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+
+import static io.trino.tests.product.launcher.cli.Commands.runCommand;
+import static java.util.Objects.requireNonNull;
+
+@CommandLine.Command(
+        name = "inspect",
```
Comment (losipiuk) [inline testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/EnvironmentInspect.java:52]: describe would be more appropriate I think

Comment (losipiuk) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/EnvironmentInspect.java:52]: inspect feels like "attach to something existing and say how it looks like"

**Review by losipiuk — APPROVED** [review]

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/EnvironmentInspect.java (hunk 2 above)
Comment (wendigo) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/EnvironmentInspect.java:53]: Sure. Makes sense :)

File: testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/EnvironmentInspect.java (hunk 1 above)
Comment (wendigo) [reply testing/trino-product-tests-launcher/src/main/java/io/trino/tests/product/launcher/cli/EnvironmentInspect.java:170]: Yes. You can't copy simlink afair

**Review by hashhar — APPROVED** [review]

**Comment by wendigo** [comment]
@hashhar please merge :)

**Pull request merged**